shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -75,12 +76,20 @@ def _apply_sum_constraint(inputs, indexes, coef_sum_max):
         currents_0 = self.initial_guess_mean_bounds()
 
         # Optimize
-        currents_sp = opt.minimize(self._residuals, currents_0,
-                                   args=(unshimmed_vec, coil_mat),
-                                   method='SLSQP',
-                                   bounds=self.merged_bounds,
-                                   constraints=tuple(constraints),
-                                   options={'maxiter': 500})
+        # When clipping to bounds, scipy raises a warning. Since this can be frequent for our purposes, we ignore that
+        # warning
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', message="Values in x were outside bounds during a "
+                                                      "minimize step, clipping to bounds",
+                                    category=RuntimeWarning,
+                                    module='scipy')
```

Comment:
When optimizing, scipy raises a warning if the optimization reaches the bounds. This causes 2 problem, one is that this raises warnings for the tests. It also clutters the output on the terminal for what is an "expected" behaviour (That's why we have bounds). I'm wondering if this is acceptable practice to silence this warning. @jcohenadad @joshuacwnewton

Reply:
usually warnings are there for a reason, but in this case I understand that it has to do with the optimization process and that it _could_ be ignored as long as the information is always irrelevant for the user.

For example: could we imagine a scenario where an optimization method would reach the bounds for many channels, and that solution would be less optimal than with another optimization method, in which case the user might want to know that they should maybe try using another optimization method?  Alternatively, as part of the output figures, you could output a graphs with all channels and their shim voltage, and showing the bounds on the graphs, so the user would have a quick overview about the number of maxed out channels (and then decide whether they want to try another method)

Reply:
My apologies, I missed this ping! I don't have prior experience with scipy's optimization tools, so it's tricky for me to comment on what practices are typical. (I also don't have much prior exposure to section of ST's codebase.)

But, if you still need another set of eyes on this, please feel free to re-ping me and I can try to dig into this warning a little more.